src/Shared/AppModels/AppDebug.cs

```diff
@@ -0,0 +1,229 @@
+            MemoryManager.AppMemoryUsageDecreased += MemoryManager_MemoryChanged;
+            UpdateMemoryUsage();
+            SetWindowSize();
+            _memoryWatchTimer = new DispatcherTimer();
+            _memoryWatchTimer.Interval = TimeSpan.FromSeconds(3);
+            _memoryWatchTimer.Tick += DispatchTimer_Elapsed;
+            _memoryWatchTimer.Start();
+        }
+
+        private void DispatchTimer_Elapsed(object sender, object e)
+        {
+            _memoryWatchTimer.Stop();
+            UpdateMemoryUsage();
+            _memoryWatchTimer.Start();
+        }
+
+        private void MemoryManager_MemoryChanged(object sender, object e)
+        {
+            UpdateMemoryUsage();
+        }
+
+        private void UpdateMemoryUsage()
+        {
+            var memoryUsed = MemoryManager.AppMemoryUsage;
+            var totalMemory = MemoryManager.AppMemoryUsageLimit;
+            MemoryUsage = SizeSuffix((long)memoryUsed);
+            TotalMemory = SizeSuffix((long)totalMemory);
+        }
+
+        private void Current_SizeChanged(object sender, Windows.UI.Core.WindowSizeChangedEventArgs e) => SetWindowSize();
+
+        private void SetWindowSize()
+        {
+            WindowHeight = Window.Current.Bounds.Height.ToString(); ;
+            WindowWidth = Window.Current.Bounds.Width.ToString();
+        }
+
+        private async void Logger_MessageReceived(object sender, LoggerMessageEventArgs e)
+        {
+            await _semaphoreSlim.WaitAsync();
+
+            var formatedMessage = LogFormatter.GetFormattedLogMessage(e);
+
+            if (AppLogs == null)
+                return;
+
+            await CoreApplication.MainView.CoreWindow.Dispatcher.RunAsync(CoreDispatcherPriority.Normal,
+            () =>
+            {
+                AppLogs.Add(formatedMessage);
+            });
+
+            _semaphoreSlim.Release();
+        }
+
+        [RelayCommand]
+        private async Task OpenLogFolderAsync()
+        {
+            var logsLocation = LogFormatter.LogFolderPath;
+            var folder = await StorageFolder.GetFolderFromPathAsync(logsLocation);
+
+            await Launcher.LaunchFolderAsync(folder);
+        }
+
+        [RelayCommand]
+        private async Task DeleteUserDataAsync()
+        {
+            _deleteUserDataDialog = new ContentDialog()
+            {
+                Title = "Delete user data",
+                Content = "Are you sure you want to delete user data? If you hit yes, the app will restart.",
+                PrimaryButtonText = "Yes",
+                SecondaryButtonText = "No",
+                PrimaryButtonCommand = PerformUserDataDeletionCommand
+            };
+
+            await _deleteUserDataDialog.ShowAsync();
+        }
+
+        [RelayCommand]
+        private async Task PerformUserDataDeletionAsync()
+        {
+            try
+            {
+                var localDataFolder = ApplicationData.Current.LocalFolder.Path;
```

Comment:
The folder being emptied needs to populated from a property on `AppDebug`, and it should be assigned the `IModifiableFolder` passed into the constructor of `AppRoot`.

In OwlCore.Storage, a folder can only be deleted if we know what to delete it from. In other words, we need a reference to the parent folder. Some folder implementations aren't addressable and have no parent, so this won't work. 

Instead, since we're given an `IModifiableFolder`, we can instead delete the _contents_ of the folder, rather than the folder itself.